mailpoet/lib/Form/DisplayFormInWPContent.php

```diff
@@ -313,4 +357,28 @@ private function shouldDisplayFormOnTag(array $setup): bool {
     if ($this->wp->hasTerm($setup['tags'], 'product_tag')) return true;
     return false;
   }
+
+  private function shouldDisplayFormOnFrontPage(array $setup): bool {
+    return ($setup['homepage'] ?? false) === '1';
+  }
+
+  private function shouldDisplayFormOnCategoryArchive($setup): bool {
+    if (!isset($setup['categoryArchives'])) return false;
+    if (($setup['categoryArchives']['all'] ?? false) === '1') return true;
+    $selectedCategories = $setup['categoryArchives']['selected'] ?? [];
+    if ($selectedCategories === []) return false;
+    return $this->wp->hasCategory($selectedCategories) || $this->wp->hasTerm($selectedCategories, 'product_cat');
```

Comment:
Just to confirm my understanding: `$this->wp->hasCategory($selectedCategories)` covers posts categories and `$this->wp->hasTerm($selectedCategories, 'product_cat')` product categories, correct?

Reply:
Exactly. I followed the example of `shouldDisplayFormOnCategory` and `shouldDisplayFormOnTag`, which work the same way, including how categories/tags are selected.